packages/core/strapi/lib/services/webhook-store.js

```diff
@@ -47,10 +50,50 @@ const fromDBObject = (row) => {
   };
 };
 
+const urlRegex =
+  /^(?:([a-z0-9+.-]+):\/\/)(?:\S+(?::\S*)?@)?(?:(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9_]-*)*[a-z\u00a1-\uffff0-9_]+)(?:\.(?:[a-z\u00a1-\uffff0-9_]-*)*[a-z\u00a1-\uffff0-9_]+)*\.?)(?::\d{2,5})?(?:[/?#]\S*)?$/;
+
+const webhookValidator = (allowedEvents) =>
+  yup
+    .object({
+      name: yup.string().required(),
+      url: yup.string().matches(urlRegex, 'url must be a valid URL').required(),
+      headers: yup.lazy((data) => {
+        if (typeof data !== 'object') {
+          return yup.object().required();
+        }
+
+        return yup
+          .object(
+            mapValues(() => {
+              yup.string().min(1).required();
+            }, data)
+          )
+          .required();
+      }),
+      events: yup.array().of(yup.string().oneOf(Array.from(allowedEvents.values())).required()),
```

Comment:
I think the main yup validation we could leave it in the controller, and leave only the evens here, wdyt?

that way we will have the flexibility to use a different contract between controller and service.